pkg/download/download.go

```diff
@@ -67,43 +70,94 @@ func Download(endpoint string, opts ...Option) (b []byte, err error) {
 		opt(dlOpts)
 	}
 
-	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
-	if err != nil {
-		return b, err
-	}
+	switch u.Scheme {
+	case "http", "https":
```

Comment:
one more option which I liked in the past is to register `tftp` as a new protocol and keep the code clean from the difference in the protocol: https://godoc.org/net/http#Transport.RegisterProtocol

Reply:
I had some for the ftp written a long time ago, it might be that `tftp` would be easier for a case of a single download (no need to keep connections around, etc): https://github.com/smira/go-ftp-protocol/blob/master/protocol/protocol.go

Reply:
👍 update on the way